postgres-driver/application.go

```diff
@@ -160,6 +151,211 @@ func (a *dbApplication) toApplication() *repository.Application {
+		ContactEmail:  newSQLNullString(app.ContactEmail),
+		Description:   newSQLNullString(app.Description),
+		Owner:         newSQLNullString(app.Owner),
+		URL:           newSQLNullString(app.URL),
+		CreatedAt:     app.CreatedAt,
+		UpdatedAt:     app.UpdatedAt,
+	}
+}
+
+type insertFreeTierAAT struct {
+	ApplicationID   string         `db:"application_id"`
+	PublicKey       sql.NullString `db:"public_key"`
+	Signature       sql.NullString `db:"signature"`
+	ClientPublicKey sql.NullString `db:"client_public_key"`
+	Version         sql.NullString `db:"version"`
+}
+
+func (i *insertFreeTierAAT) isNotNull() bool {
+	return i.PublicKey.Valid || i.Signature.Valid || i.ClientPublicKey.Valid || i.Version.Valid
+}
+
+func extractInsertFreeTierAAT(app *repository.Application) *insertFreeTierAAT {
+	return &insertFreeTierAAT{
+		ApplicationID:   app.ID,
+		PublicKey:       newSQLNullString(app.FreeTierAAT.ApplicationPublicKey),
+		Signature:       newSQLNullString(app.FreeTierAAT.ApplicationSignature),
+		ClientPublicKey: newSQLNullString(app.FreeTierAAT.ClientPublicKey),
+		Version:         newSQLNullString(app.FreeTierAAT.Version),
+	}
+}
+
+type insertGatewayAAT struct {
+	ApplicationID   string         `db:"application_id"`
+	PublicKey       sql.NullString `db:"public_key"`
+	Signature       sql.NullString `db:"signature"`
+	ClientPublicKey sql.NullString `db:"client_public_key"`
+	Version         sql.NullString `db:"version"`
+}
+
+func (i *insertGatewayAAT) isNotNull() bool {
+	return i.PublicKey.Valid || i.Signature.Valid || i.ClientPublicKey.Valid || i.Version.Valid
+}
+
+func extractInsertGatewayAAT(app *repository.Application) *insertGatewayAAT {
+	return &insertGatewayAAT{
+		ApplicationID:   app.ID,
+		PublicKey:       newSQLNullString(app.GatewayAAT.ApplicationPublicKey),
+		Signature:       newSQLNullString(app.GatewayAAT.ApplicationSignature),
+		ClientPublicKey: newSQLNullString(app.GatewayAAT.ClientPublicKey),
+		Version:         newSQLNullString(app.GatewayAAT.Version),
+	}
+}
+
+type insertFreeTierAppAccount struct {
+	ApplicationID string         `db:"application_id"`
+	PublicKey     sql.NullString `db:"public_key"`
+	Address       sql.NullString `db:"address"`
+	PrivateKey    sql.NullString `db:"private_key"`
+	Version       sql.NullString `db:"version"`
+}
+
+func (i *insertFreeTierAppAccount) isNotNull() bool {
+	return i.PublicKey.Valid || i.Address.Valid || i.PrivateKey.Valid || i.Version.Valid
+}
+
+func extractInsertFreeTierAppAccount(app *repository.Application) *insertFreeTierAppAccount {
+	return &insertFreeTierAppAccount{
+		ApplicationID: app.ID,
+		PublicKey:     newSQLNullString(app.FreeTierApplicationAccount.PublicKey),
+		Address:       newSQLNullString(app.FreeTierApplicationAccount.Address),
+		PrivateKey:    newSQLNullString(app.FreeTierApplicationAccount.PrivateKey),
+		Version:       newSQLNullString(app.FreeTierApplicationAccount.Version),
+	}
+}
+
+type insertPublicPocketAccount struct {
```

Comment:
I don't have `PublicPocketAccount` as a field for the creation of an Application, should I add it to the create application logic in PUB? 

Assuming it just uses the address and pub key created by Pocket Go's random account creation?

Reply:
This field is repeated with `freeTierApplicationAccount` for now u can just don't send it, then we can discuss if we should delete it and how.